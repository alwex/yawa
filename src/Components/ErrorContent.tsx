import React from 'react'
import style from './Styles/ErrorStyle'
import { View, Text, Button } from 'native-base'

interface ErrorContentProps {
  onRetry: () => void
}

export default class ErrorContent extends React.Component<ErrorContentProps> {
  render() {
    const { onRetry } = this.props
    return (
      <View style={style.container}>
        <Text style={style.description}>Oups!</Text>
        <View style={style.button}>
          <Button onPress={onRetry}>
            <Text>Retry</Text>
          </Button>
        </View>
      </View>
    )
  }
}
