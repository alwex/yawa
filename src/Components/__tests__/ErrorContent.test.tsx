import React from 'react'
import renderer from 'react-test-renderer'
import ErrorContent from '../ErrorContent'

describe('ErrorContent', () => {
  it('render correctly', () => {
    const component = renderer.create(<ErrorContent onRetry={() => {}} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
