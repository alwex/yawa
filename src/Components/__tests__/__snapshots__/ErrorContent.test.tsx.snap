// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorContent render correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "color": "#2C3A3F",
          "fontSize": 30,
        },
      ]
    }
    uppercase={false}
  >
    Oups!
  </Text>
  <View
    style={
      Array [
        Object {},
        Object {
          "alignSelf": "center",
          "marginTop": 20,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#007aff",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        uppercase={false}
      >
        Retry
      </Text>
    </View>
  </View>
</View>
`;
